client/src/Containers/Explorer.jsx

```diff
@@ -28,34 +37,37 @@ class Explorer extends Component {
     }
 
     async componentDidMount() {
-        let myRes = await this.loadList();
-        this.setState({ files: myRes.dirEntries.children });
-        this.setState({prevDirStruct: myRes.dirEntries.path});
+        await this.loadDir();
     }
 
-    loadList = async () => {
-        const res = await fetch(`/api/dirEntries/${this.state.curDir._id}`, {
+    loadDir = async id => {
+        const res = await fetch(`/api/dirEntries/${id || this.state.curDir._id}`, {
             method: 'GET',
             headers: { authorization: this.props.token }
-        })
-        this.setState({ dirChanges: false });
-        return await res.json();
+        });
+
+        const curDir = await res.json();
+
+        this.setState({
+          curDir,
+          selectedFiles: [],
+        });
+
+        return curDir;
```

Comment:
* `loadDir` agora recebe um `id` opcional por parâmetro. Caso um não seja passado, ele usa a ID em `state.curDir._id`.
* `loadDir` agora atualiza o estado de `curDir` e `selectedFiles` automaticamente. Isso precisa acontecer sempre que um novo diretório for carregado, em qualquer circunstância, então faz sentido isso ser implementado aqui e não em call sites (outras funções que chamam `loadDir`).

Com isto, `loadDir` pode ser usado tanto pra mudança de diretórios (usando `loadDir(newId)`) quanto recarregamento do diretório atual (com `loadDir()`).